Core/Versioning/KspVersionCriteria.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace CKAN.Versioning
+{
+    public class KspVersionCriteria
+    {
+        private List<KspVersion> versions = new List<KspVersion> ();
+
+        public KspVersionCriteria (KspVersion v)
+        {
+            this.versions.Add (v);
+        }
+
+        public KspVersionCriteria(KspVersion v, List<KspVersion> compatibleVersions)
+        {
+            if(v != null) { 
+                this.versions.Add(v);
+            }
+            this.versions.AddRange(compatibleVersions);
+        }
+
+        public List<KspVersion> Versions {
```

Comment:
>Please keep record-like classes immutable. 

It now returns `_versions.AsReadOnly()`, is it OK?

>Actually, I'd rather have this class implement IEnumerable<KspVersion>

If the name is changed to `KspVersionSet ` as you suggested then it could be wise, but I am not sure about naming change (please see my comment above).